blue_only.py

```diff
@@ -213,6 +221,165 @@ def main() -> int:
+        if not best_landing_pad:
+            continue
+
+        x, y = best_landing_pad.x_1, best_landing_pad.y_1
+        w = best_landing_pad.x_2 - best_landing_pad.x_1
+        h = best_landing_pad.y_2 - best_landing_pad.y_1
+
+        rect_image = copy.deepcopy(image)
+
+        # Draw a rectangle around the largest circular contour0
+        cv2.rectangle(rect_image, (int(x), int(y)), (int(x + w), int(y + h)), (0, 255, 0), 2)
+        x_center = x + w / 2
+        y_center = y + h / 2
+        angle_x = (x_center - im_w / 2) * (FOV_X * (math.pi / 180)) / im_w
+        angle_y = (y_center - im_h / 2) * (FOV_Y * (math.pi / 180)) / im_h
+        cv2.circle(rect_image, (int(x_center), int(y_center)), 2, (0, 0, 255), 2)
+        print("X Angle (rad): ", angle_x)
+        print("Y Angle (rad): ", angle_y)
+        target_dist = calc_target_distance(altitude_m, angle_x, angle_y)
+        is_pad_detected = True
+        vehicle.mav.landing_target_send(
+            0,
+            0,
+            pymavlink.mavutil.mavlink.MAV_FRAME_BODY_NED,
+            angle_x,
+            angle_y,
+            target_dist,
+            0,
+            0,
+        )
+
+        loop_counter += 1
+        if current_milli_time() - last_time > 1000:
+            print("FPS:", loop_counter)
+            loop_counter = 0
+            last_time = current_milli_time()
+        if current_milli_time() - last_image_time > 100:
+            if is_pad_detected:
+                print("Bounding Box Image Write")
+                cv2.imwrite(SAVE_PREFIX + "_" + str(time.time()) + ".png", rect_image)
+            else:
+                print("Plain Image Write")
+                cv2.imwrite(SAVE_PREFIX + "_" + str(time.time()) + ".png", image)
+            last_image_time = current_milli_time()
+
+    return 0
+
+
+def main() -> int:
+    """
+    Main function.
+    """
+    try:
+        with CONFIG_FILE_PATH.open("r", encoding="utf8") as file:
+            try:
+                config = yaml.safe_load(file)
+            except yaml.YAMLError as exc:
+                print(f"Error parsing YAML file: {exc}")
+                return -1
+    except FileNotFoundError:
+        print(f"File not found: {CONFIG_FILE_PATH}")
+        return -1
+    except IOError as exc:
+        print(f"Error when opening file: {exc}")
+
+    parser = argparse.ArgumentParser(description="AEAC 2024 Auto-landing script")
+    parser.add_argument(
+        "--method",
+        help="Method for selecting landing pads",
+        choices=["contour", "yolo"],
+        default="contour",
+    )
+    args = parser.parse_args()
+
+    if args.method == "contour":
+        return run_with_contours()
+
+    if args.method == "yolo":
+        model_device = config["yolo_detect_target"]["device"]
+        detect_confidence = config["yolo_detect_target"]["confidence"]
+        model_path = config["yolo_detect_target"]["model_path"]
+
+        return yolo_detect(model_device, detect_confidence, model_path)

```

Comment:
sorry I didn't mean to copy the entire script into the functions - we want to have the while loop in main, send the mavlink in main, etc. but just get the x, y, w, h from the contour or yolo function depending on the argparse. I was thinking more like - the line numbers are from the main branch and this isn't exact but like architecturally this is what I was thinking
```
def detect_landing_pads_contour():
    # basically copy paste lines 148-173
    return x, y, w, h

def detect_landing_pads_yolo():
    # implement the class and functions in yolo_detection.py to get a landing pad
    x, y, w, h = ... # interpret the results
    return x, y, w, h

def main():
    # setup stuff and argparse
    while True:
        # lines 87 - 148

          if args.method == "contour":
             x, y, w, h = detect_landing_pads_contour()
          if args.method == "yolo":
              x, y, w, h = detect_landing_pads_yolo()
          
          if x, y, w, h is None: # not exactly this but basically check that we have a bounding box
              return
          
          # lines 174 onward
```